app/main.py

```diff
@@ -1,8 +1,56 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class, clean_mark, brand):
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(
+            self,
+            distance_from_city_center: int,
+            clean_power: int,
+            average_rating: float,
+            count_of_ratings: int
+    ):
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def calculate_washing_price(self, other: Car):
+        if isinstance(other, Car):
+            difference = self.clean_power - other.clean_mark
+            product = other.comfort_class * difference * self.average_rating
+            return product / self.distance_from_city_center
+        else:
+            raise TypeError(f"'{type(other)}' not supported type. Requires 'Car'")
+
+    def wash_single_car(self, other: Car):
+        if isinstance(other, Car):
+            other.clean_mark = self.clean_power
+        else:
+            raise TypeError(f"{type(other)} type not supported. Requires 'Car'")
```

Comment:
If I want to use this method on objects what do I receive?
you need condition   `if self.clean_power > other.clean_mark:`

Reply:
Yes, I understand. I would generally make this method static. But then it will not fit into the condition of the problem. I assumed that the car will only be washed if they pay for it. In another case, I will repeat the same condition, or I will have to go through the list of machines twice.

Reply:
I understand but the task says that the function must be implemented this way